be/src/io/fs/file_system.h

```diff
@@ -64,6 +65,43 @@ struct FileInfo {
     bool is_file;
 };
 
+struct FsListGenerator {
```

Comment:
class?
and, an interface/base class should always have sufficient and clear comments for all the functions.

consider naming FsListGenerator -> FileListIterator ?